Sources/HTTP/HTTPHeadersAccessors.swift

```diff
@@ -0,0 +1,1229 @@
+            return "\(type.description) \(params)"
+        }
+    }
+    
+    /// EntryTag used in `ETag`, `If-Modified`, etc
+    public enum EntryTag: CustomStringConvertible, Equatable, Hashable {
+        /// Regular entry tag
+        case strong(String)
+        /// Weak entry tag prefixed with `"W/"`
+        case weak(String)
+        /// Equals to `"*"`
+        case wildcard
+        
+        public init(_ rawValue: String) {
+            // Check begins with W/" in case-insensitive manner to indicate is weak or not
+            if rawValue.range(of: "W/\"", options: [.anchored, .caseInsensitive]) != nil {
+                let linted = rawValue.trimmingCharacters(in: .whitespaces).replacingOccurrences(of: "W/\"", with: "", options: [.anchored, .caseInsensitive]).trimmingCharacters(in: .quotedWhitespace)
+                self = .weak(linted)
+            }
+            // Check value is wildcard
+            if rawValue == "*" {
+                self = .wildcard
+            }
+            // Value is strong
+            let linted = rawValue.trimmingCharacters(in: .whitespaces).trimmingCharacters(in: .quotedWhitespace)
+            self = .strong(linted)
+        }
+        
+        public var description: String {
+            switch self {
+            case .strong(let etag):
+                let lintedEtag = etag.trimmingCharacters(in: .quotedWhitespace)
+                return "\"\(lintedEtag)\""
+            case .weak(let etag):
+                let lintedEtag = etag.replacingOccurrences(of: "W/\"", with: "", options: [.anchored, .caseInsensitive]).trimmingCharacters(in: .quotedWhitespace)
+                return "W/\"\(lintedEtag)\""
+            case .wildcard:
+                return "*"
+            }
+        }
+        
+        public var hashValue: Int {
+            return self.description.hashValue
+        }
+        
+        public static func ==(lhs: HTTPHeaders.EntryTag, rhs: HTTPHeaders.EntryTag) -> Bool {
+            return lhs.description == rhs.description
+        }
+    }
+    
+    /// Encoding of body
+    public enum Encoding: String {
+        /// Accepting all encodings available
+        case all = "*"
+        /// Send body as is
+        case identity
+        /// Compress body data using lzw method
+        case compress
+        /// Compress body data using zlib deflate method
+        case deflate
+        /// Compress body data using gzip method
+        case gzip
+        /// Compress body data using brotli method
+        case brotli = "br"
+        
+        // These values are valid for Transfer Encoding
+        
+        /// Chunked body in `Transfer-Encoding` response header
+        case chunked
+        /// Can have trailers in `TE` request header
+        case trailers
+    }
+    
+    /// Determines server accepts `Range` header or not
+    public enum RangeType: String {
+        /// Can't accept Range
+        case none
```

Comment:
I recently read RFC7233 for implementation somewhere else. RCF allows to define any kind of range not only `bytes`. In the specific cases of File Serving `bytes` just happens to be the most common case. Ranges could be used in other REST, etc calls too and it could be `items`, `rows`, etc whatever, restricting this to only bytes might not allow http module users to use this function to other purposes than file serving.

Reply:
Please find fixed RangeType and ifRange implementations